status/queues.py

```diff
@@ -755,10 +755,12 @@ def get(self):
             sample_dict['Reagent Label'] = geno_queries.get_reagentlabel_for_sample(geno_session, sampleid)
 
             #Get udfs specific to a step
-            if stepid in step_level_udfs_id:
-                step_level_udfs = geno_queries.get_sample_udfs_from_step(geno_session, sampleid, stepid, step_level_udfs_id[stepid])
-                for udfname, udfvalue, _ in step_level_udfs:
-                    sample_dict[udfname] = udfvalue 
+            for step in stepid_to_stepindex:
+                if stepid_to_stepindex[step]<= stepid_to_stepindex[stepid]:
+                    if step in step_level_udfs_id:
```

Comment:
What's `step_level_udfs_id`?

Reply:
They are ids of steps which have associated udfs which we want to retrieve. Its defined on line 735

Reply:
There's a lot of dictionary juggling going on... A bit tricky to read I think but its ok for now.

Reply:
I agree but it was to avoid hardcoding the stepids we want from the workflow, could have been done in a different way.